File: src/Codility/Lessons/2-2_twinsArray.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(A):
    # write your code in Python 3.8.10
    if not A:
        return 0
    s = set()
    for el in A:
        if el in s:
            s.remove(el)
        else:
            s.add(el)
    for el in s:
        return el

File: src/Codility/Lessons/2-2_twinsArray.cpp
#include <iostream>
#include <vector>
#include <set>

using namespace std;

// you can use includes, for example:
// #include <algorithm>

// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

int solution(vector<int> &A)
{
    // write your code in C++14 (g++ 6.2.0)
    set<int> s;
    for (int n: A)
    {
        if (s.find(n) != s.end())
        {
            s.erase(n);
        }
        else
        {
            s.insert(n);
        }
    }

    for (int n: s)
    {
        return n;
    }
}
